src/main/java/seedu/foodrem/model/tag/UniqueTagList.java

```diff
@@ -110,7 +110,7 @@ public Iterator<Tag> iterator() {
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof UniqueTagList // instanceof handles nulls
-                && internalList.equals(((UniqueTagList) other).internalList));
+                && new HashSet<>(internalList).equals(new HashSet<>(((UniqueTagList) other).internalList)));
```

Comment:
May I know why the reason for this change? Why is there a need for the `HashSet` here?

Reply:
The ordering of the list matters while checking if they are equal. ["vege", "fruits"] will be different from ["fruits", "vege"]

Reply:
By converting it into a hash set we remove the ordering of the items in the list

Reply:
Makes sense, good catch!